mlflow/spark.py

```diff
@@ -911,24 +917,38 @@ def predict(self, pandas_df):
         spark_df = _find_and_set_features_col_as_vector_if_needed(
             self.spark.createDataFrame(pandas_df), self.spark_model
         )
-        prediction_column = "prediction"
-        if isinstance(self.spark_model, PipelineModel) and self.spark_model.stages[-1].hasParam(
-            "outputCol"
+        if (
+            self.signature is not None
+            and self.signature.outputs is not None
```

Comment:
won't `self.signature.outputs.input_names()` return `False` if `self.signature.output` is `None`?

Reply:
That would fail with `AttributeError`. 
So, in order to avoid that, I have to check for existence of both `self.signature` and `self.signature.output`